test/unit/puppetlabs/services/master/master_core_test.clj

```diff
@@ -439,3 +441,33 @@
                                                                  :facts true}}))))]
             (is (= 200 (:status response)))
             (is (= {:cool "catalog"} (json/decode (:body response) true))))))))
+
+(deftest bolt-projects-test
+  (let [bolt-projects-dir "./dev-resources/puppetlabs/services/master/master_core_test/bolt_projects"]
+    (testing "find the root of a bolt project"
+      (testing "regular project"
+        (is (= (str bolt-projects-dir "/local")
+               (get-project-root bolt-projects-dir "local"))))
+
+      (testing "embedded project (Boltdir)"
+        (is (= (str bolt-projects-dir "/embedded/Boltdir")
+               (get-project-root bolt-projects-dir "embedded")))))
+
+    (testing "finding a file in a project"
+      (testing "in the modules mount"
+        (is (= (fs/file (str bolt-projects-dir "/local/modules/helpers/files/marco.sh" ))
+               (fs/file (find-project-file bolt-projects-dir "local" "modules" "helpers" "marco.sh"))))
+        (is (= (fs/file (str bolt-projects-dir "/local/site-modules/utilities/files/etc/greeting" ))
+               (fs/file (find-project-file bolt-projects-dir "local" "modules" "utilities" "etc/greeting")))))
```

Comment:
probably should just compare strings here, instead of wrapping both sides in `fs/file`